src/modeler/modeler/Modeler.cpp

```diff
@@ -97,85 +100,99 @@ class SystemLinearProblemBuilder final
-          parameters.lastTimeStep,  // global = local
-          0};
-        SystemLinearProblemBuilder system_linear_problem(data.system.get(),
-                                                         ortools_linear_problem,
-                                                         *data.dataSeries,
-                                                         data.scenarioGroupRepository);
-
-        system_linear_problem.Provide(timeScenarioCtx);
-
-        logs.info() << "Linear problem provided";
-
-        logs.info() << "Number of variables: " << ortools_linear_problem.variableCount();
-        logs.info() << "Number of constraints: " << ortools_linear_problem.constraintCount();
-
-        measure.tick();
-        logs.info();
-        logs.info() << "Modeler build took " << measure.toStringInSeconds();
-
-        writer_.writeProblem(ortools_linear_problem);
-
-        logs.info() << "Launching resolution...";
-        measure.reset();
-        auto* solution = ortools_linear_problem.solve(parameters.solverLogs);
-        measure.tick();
-        logs.info() << "Solved in " << measure.toStringInSeconds();
-
-        switch (solution->getStatus())
-        {
-        case MipStatus::OPTIMAL:
-        case MipStatus::FEASIBLE:
-        {
-            writer_.writeSimulationTable(ortools_linear_problem,
-                                         *solution,
-                                         data,
-                                         system_linear_problem.getOptimEntityContainer(),
-                                         timeScenarioCtx);
-        }
-        break;
-        default:
-            logs.error() << "Problem during linear optimization";
-        }
+        data = loader_.loadAll();
     }
     catch (const LoadFiles::ErrorLoadingYaml&)
     {
         throw ModelerError("Error while loading files, exiting");
     }
+
+    Utils::TimeMeasurement measure;
+
+    logs.info() << "linear problem of System loaded";
+    // Problem is MIP if any variable of any component is not continuous
+    bool isMip = std::ranges::any_of(
+      data.system->Components(),
+      [](const auto& component)
+      {
+          return std::any_of(component.getModel()->Variables().cbegin(),
+                             component.getModel()->Variables().cend(),
+                             [](const auto& variable) {
+                                 return variable.Type()
+                                        != ModelerStudy::SystemModel::ValueType::FLOAT;
+                             });
+      });
+
+    // Todo: scenario
+    FillContext timeScenarioCtx = {
+      parameters.firstTimeStep,
+      parameters.lastTimeStep,
+      parameters.firstTimeStep, // global = local, single time block in pure modeler (for now)
+      parameters.lastTimeStep,  // global = local
+      0};
+
+    // Sub problem
+    auto masterAndSubPbvars = std::make_unique<MasterAndSubPbVariables>();
```

Comment:
Does your decision to use a unique_ptr (heap variable) relate to the expected large size of the object?